src/invocation/ClientConnectionManager.ts

```diff
@@ -76,32 +76,39 @@ export class ClientConnectionManager extends EventEmitter {
             this.client.getInvocationService().processResponse(data);
         };
 
-        this.triggerConnect(address).then((socket: net.Socket) => {
+        this.triggerConnect(address, asOwner).then((socket: net.Socket) => {
             let clientConnection = new ClientConnection(this.client, address, socket);
 
             return this.initiateCommunication(clientConnection).then(() => {
                 return clientConnection.registerResponseCallback(processResponseCallback);
             }).then(() => {
-                return this.authenticate(clientConnection, ownerConnection);
+                return this.authenticate(clientConnection, asOwner);
             }).then(() => {
                 this.establishedConnections[clientConnection.getAddress().toString()] = clientConnection;
                 this.onConnectionOpened(clientConnection);
                 connectionResolver.resolve(clientConnection);
             });
         }).catch((e: any) => {
-            connectionResolver.resolve(null);
+            connectionResolver.reject(e);
         }).finally(() => {
             delete this.pendingConnections[addressIndex];
         });
 
+
         let connectionTimeout = this.client.getConfig().networkConfig.connectionTimeout;
         if (connectionTimeout !== 0) {
             return connectionResolver.promise.timeout(connectionTimeout, new HazelcastError('Connection timed-out'));
         }
         return connectionResolver.promise;
     }
 
-    private triggerConnect(address: Address): Promise<net.Socket> {
+    private triggerConnect(address: Address, asOwner: boolean): Promise<net.Socket> {
+        if (!asOwner) {
+            if (this.client.getClusterService().getOwnerConnection() == null) {
+                let error = new IllegalStateError('Owner connection is not active');
```

Comment:
let us make the error string the same as Java client's: `Owner connection is not available!`